rust/tools/authorization-logic/raksha_examples/policy_patterns/NonUnanimousDecisions

```diff
@@ -0,0 +1,128 @@
+//-----------------------------------------------------------------------------
+// This gives an example with a non-unanimous decision about
+// what can be done with data belonging to multiple sources. It may be
+// non-obvious that this is possible in our system because in order
+// for data to be used, we need to prove that the owner of data allows this.
+// To accomplish this:
+//    - the data owners both delegate to a principal that represents a policy 
+//    to make the decision (using canSay)
+//    - the data owners each express their preference about what they'd like to 
+//    happen (as an attribute of the form "DataOwner" hasPreference())
+//    - The policy makes the decision using these preferences
+// Exactly how the policy makes the decision is orthogonal and can be 
+// application specific. In this case the policy is similar to role-based 
+// access control. The policy allows whichever principal occupies a specific 
+// role make the final decision.
+
+"DataOwner1" says {
+    owns("datum1").
+
+    // Delegate to a policy to decide whether or not data can be sent.
+    // The data owner opts-in to the policy by doing this delegation, even
+    // though the decision might not be DataOwner1's preference because
+    // opting in might be the only way to use this service.
+    "RoleBasedDecisionPolicy" canSay 
+        "ExampleParticle" may("send_remotely", "datum1").
+                // (this names a specific particle, but for real policies,
+                // this can be made generic).
+
+    // DataOwner1 prefers that data depending on datum1 is not sent remotely
+    // (This syntax is an attribute as described 
+    // [here](https://github.com/aferr/raksha/tree/main/rust/tools/authorization-logic#attributes-and-canactas)).
+    "DataOwner1" hasPreference("do_not_send_remotely").
+}
+
+"DataOwner2" says {
+    owns("datum2").
+    
+    "RoleBasedDecisionPolicy" canSay 
+        "ExampleParticle" may("send_remotely", "datum2").
+   
+    // Data Owner1 prefers that data depending on datum1 is not sent remotely
+    "DataOwner2" hasPreference("do_send_remotely").
+}
+
+// This is a principal that is trusted by "RoleBasedDecisonPolicy" to
+// determine if another principal can occupy a particular role. As in
+// role-based access control, whether or not a principal occupies a role
+// can be used to make access decisions. In this example the role is
+// "DeviceOwner". Other roles are also possible, and it is also possible
+// to make membership of a role conditional on other facts.
+"RoleAuthority" says {
+    "DataOwner2" canActAs "DeviceOwner".
+}
+
+
+// This is a policy that collects the preferences of both data owners
+// on whether or not they are OK with data being sent. However, the 
+// actual decision depends only on the preference of the principal
+// in a specifc role (DeviceOwner).
+
+// This role-based decision policy is one way of resolving the conflict,
+// but the pattern of delegating 
```

Comment:
Incomplete comment?

Reply:
comment completed.

Reply:
Did you commit but not push the changes here? I'm not seeing the completed comment (nor some other changes in this file).